roles/zabbix_proxy/tasks/postgresql.yml

```diff
@@ -16,77 +16,38 @@
     - database
     - dependencies
 
-
-- name: "PostgreSQL | Set the correct delegated_dbhost (to support postgres db deployment on a remote dbhost)"
-  ansible.builtin.set_fact:
-    delegated_dbhost: "{{ zabbix_proxy_dbhost if (zabbix_proxy_dbhost != 'localhost') else inventory_hostname }}"
-  when:
-    - zabbix_proxy_dbhost_run_install
-  tags:
-    - database
-
-- name: "PostgreSQL | Set the correct delegated_dbhost (to support postgres db deployment on a remote dbhost)"
-  ansible.builtin.set_fact:
-    delegated_dbhost: "{{ inventory_hostname }}"
-  when:
-    - not zabbix_proxy_dbhost_run_install
-  tags:
-    - database
-
-- name: "PostgreSQL | Delegated"
-  block:
-    - name: "PostgreSQL | Delegated | Create database"
-      community.postgresql.postgresql_db:
-        name: "{{ zabbix_proxy_dbname }}"
-        port: "{{ zabbix_proxy_dbport }}"
-        state: present
-
-    - name: "PostgreSQL | Delegated | Create database user"
-      community.postgresql.postgresql_user:
-        db: "{{ zabbix_proxy_dbname }}"
-        name: "{{ zabbix_proxy_dbuser }}"
-        password: "{{ ('md5' + (zabbix_proxy_dbpassword + zabbix_proxy_dbuser)|hash('md5')) if zabbix_proxy_dbpassword_hash_method == 'md5' else zabbix_proxy_dbpassword }}"
-        port: "{{ zabbix_proxy_dbport }}"
-        priv: ALL
-        state: present
-        encrypted: true
-  become: true
+- name: "PostgreSQL Database prep"
+  when: zabbix_proxy_database_creation
+  become: "{{ zabbix_proxy_dbhost_run_install }}"
   become_user: postgres
-  delegate_to: "{{ delegated_dbhost }}"
-  when:
-    - zabbix_proxy_database_creation
-    - zabbix_proxy_pgsql_login_host is not defined
+  delegate_to: "{{ zabbix_proxy_dbhost_run_install | ternary(delegated_dbhost, inventory_hostname) }}"
+  vars:
+    delegated_dbhost: "{{ (zabbix_server_dbhost == 'localhost') | ternary(inventory_hostname, zabbix_server_dbhost) }}"
```

Comment:
And copy-paste errors (`zabbix_server_dbhost`)  in this file aswell. Will drum up a fix. Apologies for the noise.